test/spec/services/homePagePreferenceSpec.js

```diff
@@ -0,0 +1,56 @@
+"use strict";
+
+describe("HomePagePreferenceService", function(){
+  var HomePagePreferenceService;
+  var store = {};
+
+  beforeEach(function(){
+    inject(function(_HomePagePreferenceService_){
+      HomePagePreferenceService = _HomePagePreferenceService_;
+    });
+    
+    // LocalStorage mock
+    spyOn(localStorage, 'getItem').and.callFake(function(key) {
+        return store[key];
+    });
+    
+    spyOn(localStorage, 'setItem').and.callFake(function (key, value) {
+    return store[key] = value + '';
```

Comment:
Try this to avoid the jshint error

```js
spyOn(localStorage, 'setItem').and.callFake(function (key, value) {
  /* jshint boss:true */
  return store[key] = value + '';
});
```

Reply:
Thanks Sam, updated! A little curious, do I need to change the assignment expression instead of just adding `/* jshint boss:true */` to avoid errors?

Reply:
I think what you have is fine. The linter is warning in case you meant `==` instead of `=` here.

Reply:
Thank you Sam!